### PR TITLE
refactor: move argument methods out of the Command class

### PR DESCRIPTION
**Pull request by merceyz**
**What's the problem this PR addresses?**

It's currently impossible for rollup and webpack to treeshake methods from classes so even if you're only using `Command.String` arguments all the other types will be included in the bundle

**How did you fix it?**

Move the argument methods out of the Command class

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,13 +30,13 @@ In essence you just need to declare a class that extends the `Command` abstract
 Options and command paths are set using the `Command` option declarators. Because you're in a regular class, you can easily create command that extend others! If you use TypeScript, all property types will be properly inferred with no extra work required - even your validators support coercion (cf [Typanion](https://github.com/arcanis/typanion)'s documentation for more details).
 
 ```ts
-import {Cli, Command} from 'clipanion';
+import {Cli, Command, Argument, Entries} from 'clipanion';
```
Comment (merceyz) [inline README.md:33]: Would `Commands` make more sense? :thinking:

Comment (arcanis) [reply README.md:33]: I'd go with `Builtins`

**Comment by arcanis** [comment]
I'm still not fan of `Argument` (looks too much like `arguments` imo) 🤔 What about `Option`?

Apart from that it sounds good to me - perhaps a test to ensure that the tree shaking works would help potential regression (for example if we change the build system later on).

**Comment by merceyz** [comment]
> I'm still not fan of Argument (looks too much like arguments imo) thinking What about Option?

Sounds better and I totally agree :+1: 

> perhaps a test to ensure that the tree shaking works would help potential regression (for example if we change the build system later on).

Tests added :+1:

File: tests/treeshake.test.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+      const result = await rollup({
+        input: npath.fromPortablePath(`${tempDir}/index.js`),
+        plugins: [nodeResolve()],
+      });
+
+      const code = await result.generate({ format: 'esm' });
+
+      // @ts-expect-error - matchSnapshot is added by a plugin
+      expect(code.output[0].code.replace(/\r\n?/g, '\n')).to.matchSnapshot(this);
```
Comment (arcanis) [inline tests/treeshake.test.ts:41]: I'm not sure snapshots are the right tool here, they will be very susceptible to changes 🤔

Comment (arcanis) [reply tests/treeshake.test.ts]: Updated the test to instead check the added size; should have lower noise but still catch treeshaking problems.

**Pull request merged**